react-components/intro-to-react-js.md

```diff
@@ -0,0 +1,101 @@
+# Intro to React JS
+
+## Learning Goals
+
+- What is React?
+- Why do we learn React?
+- Who is using React?
+- Intro to the Virtual DOM
+- Intro to Components
+
+## Introduction
+
+Consider a social media website, such as [Reddit.com](https://www.reddit.com/r/reactjs/). Every given web page needs to do soooo many things! To name a few, each web page needs to:
+
+- List posts, the number of upvotes, and the comments
+- Display upvote and downvote buttons
+- Handle upvote and downvote buttons
+- List related communities
+- Handle the buttons to "join" those communities
+- Display and handle the UI to create a new post
+
+![](../assets/react-components_intro-to-react-js_reddit.png)
+
+Imagine creating all of this functionality using HTML and vanilla JavaScript. It wouldn't be impossible, but our resulting code is probably:
+
+- Huge! There's a lot of code in that logic to write!
+- Unorganized. Is there one file that holds all the JS? Are there multiple files? How are they organized?
+- Full of repetition. We need to create event handlers all the time. Is there a way to simplify this repetition?
+
+Great software is maintainable, functional, and performant. How can we strive towards those values in our front-end code?
```

Comment:
I've run into this a few times lately with performant. It doesn't actually appear in the dictionary (!), and we haven't ever officially defined it in our curriculum as required jargon. It's easily avoided, so maybe rephrase?
```suggestion
Great software is maintainable, functional, and performs well. How can we strive towards those values in our front-end code?
```

Reply:
ooOOOOoh thank you! I think I end up using the word "performant" in another PR, so sorry!!! I'm taking notes.

Reply:
I don't think I even realized it wasn't a "standard" word until the spell checker yelled and me and I went looking for more info. I definitely use performant in conversation myself (and GH here isn't complaining about it), but it's definitely tech jargon.

Maybe I have a bit too prescriptive a view of word usage, but now that I know it's "non-standard" I'd be lying if I didn't say it makes me cringe a little. 🤣